napari/_app_model/injection/_processors.py

```diff
@@ -158,12 +171,27 @@ def _on_future_ready(f: Future):
     _FUTURES.add(future)
 
 
-# Add future and LayerData processors for each layer type.
+def _add_plugin_dock_widget(
+    widget_name_tuple: Tuple[Union[FunctionGui, QWidget, Widget], str],
+    viewer: viewer.Viewer = None,
+):
+    if viewer is None:
+        viewer = _provide_viewer()
+    widget, display_name = widget_name_tuple
+    viewer.window.add_dock_widget(widget, name=display_name)
+
+
 PROCESSORS: Dict[object, Callable] = {
+    # Layer type
     types.LayerDataTuple: _add_layer_data_tuples_to_viewer,
     List[types.LayerDataTuple]: _add_layer_data_tuples_to_viewer,
     layers.Layer: _add_layer_to_viewer,
+    # Dock widgets
+    Optional[
+        Tuple[Union[FunctionGui, QWidget, Widget], str]
+    ]: _add_plugin_dock_widget,
```

Comment:
This is tiny, and I'm not sure it even makes a difference, but I can't understand why this should be optional.
```suggestion
    Tuple[Union[FunctionGui, QWidget, Widget], str]: _add_plugin_dock_widget,
```

Reply:
Ah when the widget already exists, we just toggle it and return `None`

Reply:
Sorry for reopening this, I don't think it matters for functionality but my confusion is that if the widget callback returns `None` we *don't* want to run this processor (`_add_plugin_dock_wdget` will fail if the `widget_name_tuple` is `None`). Again, this seems to still work - perhaps the processors for `Optional` types just don't run on `None` anyway?